packages/expo-notifications/src/getExpoPushTokenAsync.ts

```diff
@@ -0,0 +1,182 @@
+import { Platform, CodedError } from '@unimodules/core';
+import * as Application from 'expo-application';
+import Constants from 'expo-constants';
+
+import InstallationIdProvider from './InstallationIdProvider';
+import getDevicePushTokenAsync, { DevicePushToken } from './getDevicePushTokenAsync';
+
+const productionBaseUrl = 'https://exp.host/--/api/v2/';
+
+export interface ExpoPushToken {
+  type: 'expo';
+  data: string;
+}
+
+interface Options {
+  // Endpoint URL override
+  baseUrl?: string;
+
+  // Request URL override
+  url?: string;
+
+  // Request body overrides
+  type?: string;
+  appId?: string;
+  deviceId?: string;
+  development?: boolean;
+  experienceId?: string;
+  devicePushToken?: DevicePushToken;
+}
+
+export default async function getExpoPushTokenAsync(options: Options = {}): Promise<ExpoPushToken> {
+  const devicePushToken = options.devicePushToken || (await getDevicePushTokenAsync());
+
+  const deviceId = options.deviceId || (await getDeviceIdAsync());
+
+  const experienceId =
+    options.experienceId ||
+    (Constants.manifest && `@${Constants.manifest.owner}/${Constants.manifest.slug}`);
+
+  if (!experienceId) {
+    throw new CodedError(
+      'ERR_NOTIFICATIONS_NO_EXPERIENCE_ID',
+      "No experienceId found. If it can't be inferred from the manifest (eg. in bare workflow), you have to pass it in yourself."
+    );
+  }
+
+  const appId = options.appId || Application.applicationId || experienceId;
```

Comment:
I'm not sure that falling back to `experienceId` is the right approach here 🤔 do you know if we're set up to handle that on the server? Looking at both the iOS and Android native implementations of this we only ever send the bundle/package name. I think we might want to instead throw if we can't get the `applicationId`.